pkg/network/ebpf/c/protocols/tls/sowatcher.h

```diff
@@ -54,20 +54,31 @@ static __always_inline void do_sys_open_helper_exit(exit_sys_openat_ctx *args) {
         return;
     }
 
-    // Detect whether the file being opened is a shared library
+// Check the last 9 characters of the following libraries to ensure the file is a relevant `.so`.
+// Libraries:
+//    libssl.so -> libssl.so
+// libcrypto.so -> crypto.so
+// libgnutls.so -> gnutls.so
     bool is_shared_library = false;
+#define match3chars(_base, _a,_b,_c) (path->buf[_base+i] == _a && path->buf[_base+i+1] == _b && path->buf[_base+i+2] == _c)
+#define match6chars(_base, _a,_b,_c,_d,_e,_f) (match3chars(_base,_a,_b,_c) && match3chars(_base+3,_d,_e,_f))
+    int i = 0;
 #pragma unroll
-    for (int i = 0; i < LIB_PATH_MAX_SIZE - SO_SUFFIX_SIZE; i++) {
-        if (path->buf[i] == '.' && path->buf[i + 1] == 's' && path->buf[i + 2] == 'o') {
+    for (i = 0; i < LIB_PATH_MAX_SIZE - (LIB_SO_SUFFIX_SIZE); i++) {
+        if(
+            match3chars(6, '.','s','o')) {
             is_shared_library = true;
             break;
         }
     }
-
     if (!is_shared_library) {
         goto cleanup;
     }
 
+    if (!match6chars(0, 'l','i','b','s','s','l') && !match6chars(0, 'c','r','y','p','t','o') && !match6chars(0, 'g','n','u','t','l','s')) {
+        goto cleanup;
+    }
```

Comment:
let's put documentation explaining why we split the logic into 2 parts (the scan inside the loop, and the check outside the loop)